src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -185,7 +185,14 @@ def __call__(
 
         :return: The generated token and corresponding logits
         """
+
         timer = self.timer_manager.current
+        if not timer:
```

Comment:
briefly mentioned in the code review meeting but let's implement `timer_manager.current_or_new()` so we can splash this pattern where needed

Reply:
ah gotcha. I was under the impression that we still had a `timer_manager.new_timer()` method (mentioned in the docstring) and that was what you were referring to. But yes, updating.